src/plugins/orangehrmHelpPlugin/Processor/ZendeskHelpProcessor.php

```diff
@@ -0,0 +1,118 @@
+<?php
+/**
+ * OrangeHRM is a comprehensive Human Resource Management (HRM) System that captures
+ * all the essential functionalities required for any enterprise.
+ * Copyright (C) 2006 OrangeHRM Inc., http://www.orangehrm.com
+ *
+ * OrangeHRM is free software; you can redistribute it and/or modify it under the terms of
+ * the GNU General Public License as published by the Free Software Foundation; either
+ * version 2 of the License, or (at your option) any later version.
+ *
+ * OrangeHRM is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along with this program;
+ * if not, write to the Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+ * Boston, MA 02110-1301, USA
+ */
+
+namespace OrangeHRM\Help\Processor;
+
+use Exception;
+use GuzzleHttp\Client;
+use GuzzleHttp\RequestOptions;
+use OrangeHRM\Help\Service\HelpConfigService;
+
+class ZendeskHelpProcessor implements HelpProcessor
+{
+    public const DEFAULT_CONTENT_TYPE = "application/json";
+    public const ZENDESK_SEARCH_URL = '/api/v2/help_center/articles/search.json?';
+    public const ZENDESK_DEFAULT_URL_PATH = '/hc/en-us';
+
+    protected ?HelpConfigService $helpConfigService = null;
+    private ?Client $httpClient = null;
+
+    /**
+     * @return HelpConfigService
+     */
+    public function getHelpConfigService(): HelpConfigService
+    {
+        return $this->helpConfigService ??= new HelpConfigService();
+    }
+
+    public function getHttpClient(): Client
+    {
+        return $this->httpClient ??= new Client();
+    }
+
+    /**
+     * @return String
+     */
+    public function getBaseUrl(): string
+    {
+        return $this->getHelpConfigService()->getBaseHelpUrl();
+    }
+
+    /**
+     * @param string $label
+     * @return string
+     */
+    public function getSearchUrl(string $label): string
+    {
+        return $this->getBaseUrl() . self::ZENDESK_SEARCH_URL . 'label_names=' . $label;
+    }
+
+    /**
+     * @param string $label
+     * @return string
+     */
+    public function getRedirectUrl(string $label): string
+    {
+        $searchUrl = $this->getSearchUrl($label);
+
+        $results = $this->sendQuery($searchUrl);
+        if ($results['response']) {
```

Comment:
```suggestion
        if (isset($results['response'])) {
```
if `$results` null then may fail without isset